### PR TITLE
platform: imx: Fix PLATFORM_DCACHE_ALIGN

### PR DESCRIPTION
**Pull request by dbaluta**
At this point PLATFORM_DCACHE_ALIGN is equal with DCACHE_LINE_SIZE
which on i.MX boards is 128.

Functions like rmalloc and rzalloc are allocating memory aligned at
PLATFORM_DCACHE_ALIGN bytes. This means that blocks under 128 bytes
are not really used and blocks sized > 128 are preffered for small
allocations.

This seems to be an unnecessary overhead which results in SOF not
being abel to allocate memory for simple pipelines (e.g simple
pipeline with a mixer).

For this reason, we set PLATFORM_DCACHE_ALIGN to a smaller value
(sizeof(void*)) but keep HEAP_BUF_ALIGNMENT unchanged.

Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>

### CODE REVIEW

File: src/platform/imx8/include/platform/lib/memory.h
Comment (paulstelian97) [inline src/platform/imx8/include/platform/lib/memory.h:17]: Does this build properly? This header might possibly be included by stuff that isn't C code I think...

**Review by paulstelian97 — APPROVED** [review]

File: src/platform/imx8/include/platform/lib/memory.h
Comment (dbaluta) [reply src/platform/imx8/include/platform/lib/memory.h:17]: @paulstelian97 thanks Paul very good point! This builds properly! You can see that only HEAP_BUF_ALIGNMENT is used from non C code (e.g linker files). This is also the way intel non-cavs platforms are handling the alignment.

**Comment by dbaluta** [comment]
@lgirdwood if you want you may approve this but please do not merge it. While this fixes our problems on sof-imx-stable-v1.5, i see some errors on master branch.

I will merge this PR once I figure out what's happening.

**Comment by dbaluta** [comment]
@lgirdwood this is good to merge. the problem was with the topology file and the mixer. It looks like there is something very fishy with the mixer :)

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**
